examples/advanced_operations/add_dynamic_page_feed.py

```diff
@@ -0,0 +1,364 @@
+    except GoogleAdsException as ex:
+        print('Request with ID "{}" failed with status "{}" and includes the '
+              'following errors:'.format(ex.request_id, ex.error.code().name))
+        for error in ex.failure.errors:
+            print('\tError with message "{}".'.format(error.message))
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print('\t\tOn field: {}'.format(
+                        field_path_element.field_name))
+        sys.exit(1)
+
+
+def create_feed(client, customer_id):
+    """Creates a page feed with URLs
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID str.
+
+    Returns:
+        A FeedDetails instance with information about the newly created feed.
+    """
+    # Retrieve a new feed operation object.
+    feed_operation = client.get_type('FeedOperation', version='v1')
+    # Create a new feed.
+    feed = feed_operation.create
+    feed.name.value = 'DSA Feed #{}'.format(uuid.uuid4())
+    feed.origin = client.get_type('FeedOriginEnum', version='v1').USER
+
+    feed_attribute_type_enum = client.get_type('FeedAttributeTypeEnum',
+                                               version='v1')
+
+    # Create the feed's attributes.
+    feed_attribute_url = feed.attributes.add()
+    feed_attribute_url.type = feed_attribute_type_enum.URL_LIST
+    feed_attribute_url.name.value = 'Page URL'
+
+    feed_attribute_label = feed.attributes.add()
+    feed_attribute_label.type = feed_attribute_type_enum.STRING_LIST
+    feed_attribute_label.name.value = 'Label'
+
+    # Retrieve the feed service.
+    feed_service = client.get_service('FeedService', version='v1')
+    # Send the feed operation and add the feed.
+    response = feed_service.mutate_feeds(customer_id, [feed_operation])
+
+    return response.results[0].resource_name
+
+
+def get_feed_details(client, customer_id, resource_name):
+    """Makes a search request to retrieve the attributes of a single feed.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID str.
+        resource_name: the str resource_name of a feed.
+
+    Returns:
+        A FeedDetails instance with information about the feed that was
+        retrieved in the search request.
+    """
+    query = '''
+        SELECT
+            feed.attributes
+        FROM
+            feed
+        WHERE
+            feed.resource_name = "{}"
+        LIMIT
+          1
+    '''.format(resource_name)
+
+    ga_service = client.get_service('GoogleAdsService', version='v1')
+    response = ga_service.search(customer_id, query=query)
+
+    # Maps specific fields in each row in the response to a dict. This would
+    # overwrite the same fields in the dict for each row, but we know we'll
+    # only one row will be returned.
+    for row in response:
```

Comment:
given you are limiting the response to one row, you can use .next here instead of a for loop

Reply:
Unfortunately the `GRPCIterator` objects returned by `search` aren't normal Python iterators and don't have a `__next__` method so passing them to `next` results in:

`*** TypeError: 'GRPCIterator' object is not an iterator`

They do have a `_next_page` method but that also returns a `GRPCIterator`